packages/vuetify/src/util/__tests__/makeProps.spec.ts

```diff
@@ -0,0 +1,109 @@
+import * as vue from 'vue'
+import { mount } from '@vue/test-utils'
+import * as framework from '../../framework'
+import makeProps from '../makeProps'
+
+describe('makeProps', () => {
+  it('should use global default if defined', () => {
+    // @ts-expect-error
+    jest.spyOn(vue, 'getCurrentInstance').mockReturnValueOnce({ type: { name: 'Foo' } })
+
+    jest.spyOn(framework, 'useVuetify').mockReturnValueOnce({
+      defaults: {
+        Foo: {
+          bar: 'hello world',
+        },
+      },
+    })
+
+    const props = makeProps({
+      bar: {
+        type: String,
+        default: 'goodbye world',
+      },
+    })
+
+    // @ts-expect-error
+    expect(props.bar.default()).toBe('hello world')
+  })
+
+  it('should use component default if global not defined', () => {
+    // @ts-expect-error
+    jest.spyOn(vue, 'getCurrentInstance').mockReturnValueOnce({ type: { name: 'Foo' } })
+
+    jest.spyOn(framework, 'useVuetify').mockReturnValueOnce({
+      defaults: {},
+    })
+
+    const props = makeProps({
+      bar: {
+        type: String,
+        default: 'goodbye world',
+      },
+    })
+
+    // @ts-expect-error
+    expect(props.bar.default()).toBe('goodbye world')
+  })
+
+  it('should handle prop with default factory function', () => {
+    const vuetify = framework.createVuetify({
+      defaults: {
+        TestComponent: {
+          foo: () => ({ bar: 'baz' }),
+        },
+      },
+    })
+
+    const TestComponent = vue.defineComponent({
```

Comment:
This may be a stretch but couldn't `line 80` combine with this?

Reply:
Not sure what you mean